diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index 4f4600c..e849006 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -123,17 +123,9 @@ class ProgrammerController extends BaseController
      */
     private function handleRequest(Request $request, Programmer $programmer)
     {
-        $data = json_decode($request->getContent(), true);
+        $data = $this->decodeRequestBodyIntoParameters($request);
         $isNew = !$programmer->id;
 
-        if ($data === null) {
-            $problem = new ApiProblem(
-                400,
-                ApiProblem::TYPE_INVALID_REQUEST_BODY_FORMAT
-            );
-            throw new ApiProblemException($problem);
-        }
-
         // determine which properties should be changeable on this request
         $apiProperties = array('avatarNumber', 'tagLine');
         if ($isNew) {
@@ -143,12 +135,11 @@ class ProgrammerController extends BaseController
         // update the properties
         foreach ($apiProperties as $property) {
             // if a property is missing on PATCH, that's ok - just skip it
-            if (!isset($data[$property]) && $request->isMethod('PATCH')) {
+            if (!$data->has($property) && $request->isMethod('PATCH')) {
                 continue;
             }
 
-            $val = isset($data[$property]) ? $data[$property] : null;
-            $programmer->$property = $val;
+            $programmer->$property = $data->get($property);
         }
 
         $programmer->userId = $this->getLoggedInUser()->id;
diff --git a/src/KnpU/CodeBattle/Controller/Api/TokenController.php b/src/KnpU/CodeBattle/Controller/Api/TokenController.php
index d582a6e..3569b9c 100644
--- a/src/KnpU/CodeBattle/Controller/Api/TokenController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/TokenController.php
@@ -21,7 +21,7 @@ class TokenController extends BaseController
         $data = $this->decodeRequestBodyIntoParameters($request);
 
         $token = new ApiToken($this->getLoggedInUser()->id);
-        $token->notes = $data['notes'];
+        $token->notes = $data->get('notes');
 
         $this->getApiTokenRepository()->save($token);
 
diff --git a/src/KnpU/CodeBattle/Controller/BaseController.php b/src/KnpU/CodeBattle/Controller/BaseController.php
index 08f29cb..bbead86 100644
--- a/src/KnpU/CodeBattle/Controller/BaseController.php
+++ b/src/KnpU/CodeBattle/Controller/BaseController.php
@@ -10,6 +10,7 @@ use KnpU\CodeBattle\Application;
 use Silex\Application as SilexApplication;
 use Silex\ControllerCollection;
 use Silex\ControllerProviderInterface;
+use Symfony\Component\HttpFoundation\ParameterBag;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
@@ -19,6 +20,8 @@ use KnpU\CodeBattle\Repository\ProgrammerRepository;
 use KnpU\CodeBattle\Repository\ProjectRepository;
 use KnpU\CodeBattle\Security\Token\ApiTokenRepository;
 use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use KnpU\CodeBattle\Api\ApiProblem;
+use KnpU\CodeBattle\Api\ApiProblemException;
 
 /**
  * Base controller class to hide Silex-related implementation details
@@ -266,6 +269,21 @@ abstract class BaseController implements ControllerProviderInterface
 
     protected function decodeRequestBodyIntoParameters(Request $request)
     {
-        return json_decode($request->getContent(), true);
+        // allow for a possibly empty body
+        if (!$request->getContent()) {
+            $data = array();
+        } else {
+            $data = json_decode($request->getContent(), true);
+
+            if ($data === null) {
+                $problem = new ApiProblem(
+                    400,
+                    ApiProblem::TYPE_INVALID_REQUEST_BODY_FORMAT
+                );
+                throw new ApiProblemException($problem);
+            }
+        }
+
+        return new ParameterBag($data);
     }
 }
