diff --git a/features/api/token.feature b/features/api/token.feature
index 3569c60..7cafe2e 100644
--- a/features/api/token.feature
+++ b/features/api/token.feature
@@ -28,4 +28,4 @@ Feature: Token
     And I authenticate with user "weaverryan" and password "WRONG"
     When I request "POST /api/tokens"
     Then the response status code should be 401
-    And the "detail" property should equal "Bad Credentials"
+    And the "detail" property should equal "Invalid credentials."
diff --git a/src/KnpU/CodeBattle/Controller/Api/TokenController.php b/src/KnpU/CodeBattle/Controller/Api/TokenController.php
index 9327b4d..b115d58 100644
--- a/src/KnpU/CodeBattle/Controller/Api/TokenController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/TokenController.php
@@ -7,6 +7,7 @@ use Silex\ControllerCollection;
 use Symfony\Component\HttpFoundation\Request;
 use KnpU\CodeBattle\Security\Token\ApiToken;
 use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
 
 class TokenController extends BaseController
 {
@@ -18,8 +19,12 @@ class TokenController extends BaseController
     public function newAction(Request $request)
     {
         $username = $request->headers->get('PHP_AUTH_USER');
+        $password = $request->headers->get('PHP_AUTH_PW');
 
         $user = $this->getUserRepository()->findUserByUsername($username);
+        if (!$user || !$this->isPasswordValid($user, $password)) {
+            throw new BadCredentialsException();
+        }
 
         $data = json_decode($request->getContent(), true);
 
