diff --git a/features/api/token.feature b/features/api/token.feature
index 2253b2f..8402ed2 100644
--- a/features/api/token.feature
+++ b/features/api/token.feature
@@ -28,3 +28,10 @@ Feature: Token
     And I authenticate with user "weaverryan" and password "WRONG"
     When I request "POST /api/tokens"
     Then the response status code should be 401
+
+  Scenario: Creating a token without a note
+    Given there is a user "weaverryan" with password "test"
+    And I authenticate with user "weaverryan" and password "test"
+    When I request "POST /api/tokens"
+    Then the response status code should be 400
+    And the "errors.notes" property should equal "Please add some notes about this token"
diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index e849006..61bad86 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -144,15 +144,4 @@ class ProgrammerController extends BaseController
 
         $programmer->userId = $this->getLoggedInUser()->id;
     }
-
-    private function throwApiProblemValidationException(array $errors)
-    {
-        $apiProblem = new ApiProblem(
-            400,
-            ApiProblem::TYPE_VALIDATION_ERROR
-        );
-        $apiProblem->set('errors', $errors);
-
-        throw new ApiProblemException($apiProblem);
-    }
 }
diff --git a/src/KnpU/CodeBattle/Controller/Api/TokenController.php b/src/KnpU/CodeBattle/Controller/Api/TokenController.php
index 3569b9c..51d7298 100644
--- a/src/KnpU/CodeBattle/Controller/Api/TokenController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/TokenController.php
@@ -23,6 +23,11 @@ class TokenController extends BaseController
         $token = new ApiToken($this->getLoggedInUser()->id);
         $token->notes = $data->get('notes');
 
+        $errors = $this->validate($token);
+        if ($errors) {
+            $this->throwApiProblemValidationException($errors);
+        }
+
         $this->getApiTokenRepository()->save($token);
 
         return $this->createApiResponse($token, 201);
diff --git a/src/KnpU/CodeBattle/Controller/BaseController.php b/src/KnpU/CodeBattle/Controller/BaseController.php
index bbead86..a9ebfe9 100644
--- a/src/KnpU/CodeBattle/Controller/BaseController.php
+++ b/src/KnpU/CodeBattle/Controller/BaseController.php
@@ -286,4 +286,15 @@ abstract class BaseController implements ControllerProviderInterface
 
         return new ParameterBag($data);
     }
+
+    protected function throwApiProblemValidationException(array $errors)
+    {
+        $apiProblem = new ApiProblem(
+            400,
+            ApiProblem::TYPE_VALIDATION_ERROR
+        );
+        $apiProblem->set('errors', $errors);
+
+        throw new ApiProblemException($apiProblem);
+    }
 }
