diff --git a/features/api/token.feature b/features/api/token.feature
index 3c2edca..2253b2f 100644
--- a/features/api/token.feature
+++ b/features/api/token.feature
@@ -16,3 +16,15 @@ Feature: Token
     Then the response status code should be 201
     # And the "Location" header should exist
     And the "token" property should be a string
+
+  Scenario: Creating a token with a bad password
+    Given there is a user "weaverryan" with password "test"
+    And I have the payload:
+      """
+      {
+        "notes": "A testing token!"
+      }
+      """
+    And I authenticate with user "weaverryan" and password "WRONG"
+    When I request "POST /api/tokens"
+    Then the response status code should be 401
