diff --git a/features/api/token.feature b/features/api/token.feature
index 4bb57ed..3c2edca 100644
--- a/features/api/token.feature
+++ b/features/api/token.feature
@@ -2,3 +2,17 @@ Feature: Token
   In order to access restricted information
   As an API client
   I can create tokens and use them to access information
+
+  Scenario: Creating a token
+    Given there is a user "weaverryan" with password "test"
+    And I have the payload:
+      """
+      {
+        "notes": "A testing token!"
+      }
+      """
+    And I authenticate with user "weaverryan" and password "test"
+    When I request "POST /api/tokens"
+    Then the response status code should be 201
+    # And the "Location" header should exist
+    And the "token" property should be a string
