diff --git a/src/KnpU/CodeBattle/Controller/Api/TokenController.php b/src/KnpU/CodeBattle/Controller/Api/TokenController.php
index d504dee..9c51223 100644
--- a/src/KnpU/CodeBattle/Controller/Api/TokenController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/TokenController.php
@@ -29,4 +29,5 @@ class TokenController extends BaseController
 
         return $this->createApiResponse($token, 201);
     }
+    
 }
