diff --git a/src/KnpU/CodeBattle/Controller/Api/TokenController.php b/src/KnpU/CodeBattle/Controller/Api/TokenController.php
index 9327b4d..37b091e 100644
--- a/src/KnpU/CodeBattle/Controller/Api/TokenController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/TokenController.php
@@ -7,6 +7,7 @@ use Silex\ControllerCollection;
 use Symfony\Component\HttpFoundation\Request;
 use KnpU\CodeBattle\Security\Token\ApiToken;
 use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
 
 class TokenController extends BaseController
 {
@@ -17,13 +18,11 @@ class TokenController extends BaseController
 
     public function newAction(Request $request)
     {
-        $username = $request->headers->get('PHP_AUTH_USER');
-
-        $user = $this->getUserRepository()->findUserByUsername($username);
+        $this->enforceUserSecurity();
 
         $data = json_decode($request->getContent(), true);
 
-        $token = new ApiToken($user->id);
+        $token = new ApiToken($this->getLoggedInUser()->id);
         $token->notes = $data['notes'];
 
         $this->getApiTokenRepository()->save($token);
