diff --git a/src/KnpU/CodeBattle/Controller/Api/TokenController.php b/src/KnpU/CodeBattle/Controller/Api/TokenController.php
index 9327b4d..8c1142b 100644
--- a/src/KnpU/CodeBattle/Controller/Api/TokenController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/TokenController.php
@@ -7,6 +7,7 @@ use Silex\ControllerCollection;
 use Symfony\Component\HttpFoundation\Request;
 use KnpU\CodeBattle\Security\Token\ApiToken;
 use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
 
 class TokenController extends BaseController
 {
@@ -18,8 +19,12 @@ class TokenController extends BaseController
     public function newAction(Request $request)
     {
         $username = $request->headers->get('PHP_AUTH_USER');
+        $password = $request->headers->get('PHP_AUTH_PW');
 
         $user = $this->getUserRepository()->findUserByUsername($username);
+        if (!$user || !$this->isPasswordValid($user, $password)) {
+            throw new AccessDeniedException('Bad Credentials');
+        }
 
         $data = json_decode($request->getContent(), true);
 
