diff --git a/src/KnpU/CodeBattle/Application.php b/src/KnpU/CodeBattle/Application.php
index 33dd97f..0f03000 100644
--- a/src/KnpU/CodeBattle/Application.php
+++ b/src/KnpU/CodeBattle/Application.php
@@ -239,6 +239,7 @@ class Application extends SilexApplication
                     }),
                     'stateless' => true,
                     'anonymous' => true,
+                    'http' => true,
                     'api_token' => true,
                 ),
                 'main' => array(
diff --git a/src/KnpU/CodeBattle/Controller/Api/TokenController.php b/src/KnpU/CodeBattle/Controller/Api/TokenController.php
index d504dee..fbcc24f 100644
--- a/src/KnpU/CodeBattle/Controller/Api/TokenController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/TokenController.php
@@ -16,13 +16,11 @@ class TokenController extends BaseController
 
     public function newAction(Request $request)
     {
-        $username = $request->headers->get('PHP_AUTH_USER');
-
-        $user = $this->getUserRepository()->findUserByUsername($username);
+        $this->enforceUserSecurity();
 
         $data = json_decode($request->getContent(), true);
 
-        $token = new ApiToken($user->id);
+        $token = new ApiToken($this->getLoggedInUser()->id);
         $token->notes = $data['notes'];
 
         $this->getApiTokenRepository()->save($token);
