diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index fb3cdf3..4f4600c 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -36,9 +36,7 @@ class ProgrammerController extends BaseController
 
     public function newAction(Request $request)
     {
-        if (!$this->isUserLoggedIn()) {
-            throw new AccessDeniedException();
-        }
+        $this->enforceUserSecurity();
 
         $programmer = new Programmer();
         $this->handleRequest($request, $programmer);
@@ -89,6 +87,8 @@ class ProgrammerController extends BaseController
             $this->throw404('Oh no! This programmer has deserted! We\'ll send a search party!');
         }
 
+        $this->enforceProgrammerOwnershipSecurity($programmer);
+
         $this->handleRequest($request, $programmer);
 
         if ($errors = $this->validate($programmer)) {
@@ -106,6 +106,8 @@ class ProgrammerController extends BaseController
     {
         $programmer = $this->getProgrammerRepository()->findOneByNickname($nickname);
 
+        $this->enforceProgrammerOwnershipSecurity($programmer);
+
         if ($programmer) {
             $this->delete($programmer);
         }
diff --git a/src/KnpU/CodeBattle/Controller/BaseController.php b/src/KnpU/CodeBattle/Controller/BaseController.php
index 9d6cc57..7a0bbd0 100644
--- a/src/KnpU/CodeBattle/Controller/BaseController.php
+++ b/src/KnpU/CodeBattle/Controller/BaseController.php
@@ -18,6 +18,7 @@ use Symfony\Component\HttpFoundation\Request;
 use KnpU\CodeBattle\Repository\ProgrammerRepository;
 use KnpU\CodeBattle\Repository\ProjectRepository;
 use KnpU\CodeBattle\Security\Token\ApiTokenRepository;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
 
 /**
  * Base controller class to hide Silex-related implementation details
@@ -246,4 +247,21 @@ abstract class BaseController implements ControllerProviderInterface
 
         return $this->container['serializer']->serialize($data, $format, $context);
     }
+
+    protected function enforceUserSecurity()
+    {
+        if (!$this->isUserLoggedIn()) {
+            throw new AccessDeniedException();
+        }
+    }
+
+    protected function enforceProgrammerOwnershipSecurity(Programmer $programmer)
+    {
+        $this->enforceUserSecurity();
+
+        if ($this->getLoggedInUser()->id != $programmer->userId) {
+            throw new AccessDeniedException();
+        }
+    }
+
 }
