diff --git a/src/KnpU/CodeBattle/Controller/Api/TokenController.php b/src/KnpU/CodeBattle/Controller/Api/TokenController.php
index a3efe28..d504dee 100644
--- a/src/KnpU/CodeBattle/Controller/Api/TokenController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/TokenController.php
@@ -4,6 +4,8 @@ namespace KnpU\CodeBattle\Controller\Api;
 
 use KnpU\CodeBattle\Controller\BaseController;
 use Silex\ControllerCollection;
+use Symfony\Component\HttpFoundation\Request;
+use KnpU\CodeBattle\Security\Token\ApiToken;
 
 class TokenController extends BaseController
 {
@@ -12,8 +14,19 @@ class TokenController extends BaseController
         $controllers->post('/api/tokens', array($this, 'newAction'));
     }
 
-    public function newAction()
+    public function newAction(Request $request)
     {
-        return 'foo';
+        $username = $request->headers->get('PHP_AUTH_USER');
+
+        $user = $this->getUserRepository()->findUserByUsername($username);
+
+        $data = json_decode($request->getContent(), true);
+
+        $token = new ApiToken($user->id);
+        $token->notes = $data['notes'];
+
+        $this->getApiTokenRepository()->save($token);
+
+        return $this->createApiResponse($token, 201);
     }
 }
