diff --git a/features/api/battle.feature b/features/api/battle.feature
index a1a9288..3c3bbb5 100644
--- a/features/api/battle.feature
+++ b/features/api/battle.feature
@@ -33,3 +33,4 @@ Feature:
     Then the response status code should be 200
     And the "battles" property should be an array
     And the "battles" property should contain 2 items
+    And the "battles.0.programmerUri" property should equal "/api/programmers/Fred"
diff --git a/src/KnpU/CodeBattle/Model/Battle.php b/src/KnpU/CodeBattle/Model/Battle.php
index 081415a..5c25bb5 100644
--- a/src/KnpU/CodeBattle/Model/Battle.php
+++ b/src/KnpU/CodeBattle/Model/Battle.php
@@ -2,6 +2,8 @@
 
 namespace KnpU\CodeBattle\Model;
 
+use JMS\Serializer\Annotation as Serializer;
+
 class Battle
 {
     /* All public properties are persisted */
@@ -22,4 +24,13 @@ class Battle
     public $foughtAt;
 
     public $notes;
+
+    /**
+     * @Serializer\VirtualProperty()
+     */
+    public function getProgrammerUri()
+    {
+        // todo - just faking this
+        return '/api/programmers/Fred';
+    }
 }
