diff --git a/src/KnpU/CodeBattle/Controller/Api/BattleController.php b/src/KnpU/CodeBattle/Controller/Api/BattleController.php
index 1e05fb6..9a105b7 100644
--- a/src/KnpU/CodeBattle/Controller/Api/BattleController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/BattleController.php
@@ -11,6 +11,7 @@ class BattleController extends BaseController
     protected function addRoutes(ControllerCollection $controllers)
     {
         $controllers->post('/api/battles', array($this, 'newAction'));
+        $controllers->get('/api/battles/{id}', array($this, 'showAction'));
     }
 
     public function newAction(Request $request)
@@ -46,4 +47,16 @@ class BattleController extends BaseController
 
         return $response;
     }
+
+    public function showAction($id)
+    {
+        $battle = $this->getBattleRepository()->find($id);
+        if (!$battle) {
+            $this->throw404('No battle with id '.$id);
+        }
+
+        $response = $this->createApiResponse($battle, 200);
+
+        return $response;
+    }
 }
