diff --git a/src/KnpU/CodeBattle/Controller/Api/BattleController.php b/src/KnpU/CodeBattle/Controller/Api/BattleController.php
index 8a0e347..72fbc4e 100644
--- a/src/KnpU/CodeBattle/Controller/Api/BattleController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/BattleController.php
@@ -4,6 +4,7 @@ namespace KnpU\CodeBattle\Controller\Api;
 
 use KnpU\CodeBattle\Controller\BaseController;
 use Silex\ControllerCollection;
+use Symfony\Component\HttpFoundation\Request;
 
 class BattleController extends BaseController
 {
@@ -12,8 +13,23 @@ class BattleController extends BaseController
         $controllers->post('/api/battles', array($this, 'newAction'));
     }
 
-    public function newAction()
+    public function newAction(Request $request)
     {
-        return 'foo';
+        $this->enforceUserSecurity();
+
+        $data = $this->decodeRequestBodyIntoParameters($request);
+
+        $programmerId = $data->get('programmerId');
+        $projectId = $data->get('projectId');
+
+        $programmer = $this->getProgrammerRepository()->find($programmerId);
+        $project = $this->getProjectRepository()->find($projectId);
+
+        $battle = $this->getBattleManager()->battle($programmer, $project);
+
+        $response = $this->createApiResponse($battle, 201);
+        $response->headers->set('Location', 'TODO');
+
+        return $response;
     }
 }
