diff --git a/src/KnpU/CodeBattle/Controller/Api/BattleController.php b/src/KnpU/CodeBattle/Controller/Api/BattleController.php
index 9a105b7..10fc9fa 100644
--- a/src/KnpU/CodeBattle/Controller/Api/BattleController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/BattleController.php
@@ -11,7 +11,8 @@ class BattleController extends BaseController
     protected function addRoutes(ControllerCollection $controllers)
     {
         $controllers->post('/api/battles', array($this, 'newAction'));
-        $controllers->get('/api/battles/{id}', array($this, 'showAction'));
+        $controllers->get('/api/battles/{id}', array($this, 'showAction'))
+            ->bind('api_battle_show');
     }
 
     public function newAction(Request $request)
@@ -43,7 +44,8 @@ class BattleController extends BaseController
         $battle = $this->getBattleManager()->battle($programmer, $project);
 
         $response = $this->createApiResponse($battle, 201);
-        $response->headers->set('Location', 'TODO');
+        $url = $this->generateUrl('api_battle_show', array('id' => $battle->id));
+        $response->headers->set('Location', $url);
 
         return $response;
     }
