diff --git a/features/api/battle.feature b/features/api/battle.feature
index 8874887..b737a19 100644
--- a/features/api/battle.feature
+++ b/features/api/battle.feature
@@ -31,4 +31,4 @@ Feature:
     Then the response status code should be 200
     And the "notes" property should exist
     And the "didProgrammerWin" property should exist
-    And print last response
+    And the "programmerUri" property should equal "/api/programmers/Fred"
diff --git a/src/KnpU/CodeBattle/Model/Battle.php b/src/KnpU/CodeBattle/Model/Battle.php
index 6ca1ba1..eaf2686 100644
--- a/src/KnpU/CodeBattle/Model/Battle.php
+++ b/src/KnpU/CodeBattle/Model/Battle.php
@@ -38,4 +38,12 @@ class Battle
      * @Serializer\Expose()
      */
     public $notes;
+
+    /**
+     * @Serializer\VirtualProperty()
+     */
+    public function getProgrammerUri()
+    {
+        return '/api/programmers/'.$this->programmer->nickname;
+    }
 }
