diff --git a/features/api/programmer.feature b/features/api/programmer.feature
index 69b1fe8..90b9792 100644
--- a/features/api/programmer.feature
+++ b/features/api/programmer.feature
@@ -7,3 +7,14 @@ Feature: Programmer
     Given the user "weaverryan" exists
 
   Scenario: Create a programmer
+    Given I have the payload:
+      """
+      {
+        "nickname": "ObjectOrienter555",
+        "avatarNumber" : "2"
+      }
+      """
+    When I request "POST /api/programmers"
+    Then the response status code should be 201
+    And the "Location" header should be "/api/programmers/ObjectOrienter555"
+    And the "nickname" property equals "ObjectOrienter555"
