diff --git a/features/api/ApiFeatureContext.php b/features/api/ApiFeatureContext.php
index 44d3a55..8848e8d 100644
--- a/features/api/ApiFeatureContext.php
+++ b/features/api/ApiFeatureContext.php
@@ -123,6 +123,14 @@ class ApiFeatureContext extends BehatContext
     }
 
     /**
+     * @BeforeScenario
+     */
+    public function clearData()
+    {
+        $this->getProjectHelper()->reloadDatabase();
+    }
+
+    /**
      * @Given /^I have the payload:$/
      */
     public function iHaveThePayload(PyStringNode $requestPayload)
diff --git a/features/api/programmer.feature b/features/api/programmer.feature
index b10336b..a25a42b 100644
--- a/features/api/programmer.feature
+++ b/features/api/programmer.feature
@@ -4,7 +4,7 @@ Feature: Programmer
   I need to be able to create programmers and power them up
 
   Background:
-    # Given the user "weaverryan" exists
+    Given the user "weaverryan" exists
 
   Scenario: Create a programmer
     Given I have the payload:
