diff --git a/features/api/programmer.feature b/features/api/programmer.feature
index a25a42b..1f0c4dc 100644
--- a/features/api/programmer.feature
+++ b/features/api/programmer.feature
@@ -19,3 +19,28 @@ Feature: Programmer
     Then the response status code should be 201
     And the "Location" header should be "/api/programmers/ObjectOrienter"
     And the "nickname" property should equal "ObjectOrienter"
+
+  Scenario: GET one programmer
+    Given the following programmers exist:
+      | nickname   | avatarNumber |
+      | UnitTester | 3            |
+    When I request "GET /api/programmers/UnitTester"
+    Then the response status code should be 200
+    And the following properties should exist:
+      """
+      nickname
+      avatarNumber
+      powerLevel
+      tagLine
+      """
+    And the "nickname" property should equal "UnitTester"
+
+  Scenario: GET a collection of programmers
+    Given the following programmers exist:
+      | nickname    | avatarNumber |
+      | UnitTester  | 3            |
+      | CowboyCoder | 5            |
+    When I request "GET /api/programmers"
+    Then the response status code should be 200
+    And the "programmers" property should be an array
+    And the "programmers" property should contain 2 items
