diff --git a/src/KnpU/CodeBattle/Tests/ProgrammerControllerTest.php b/src/KnpU/CodeBattle/Tests/ProgrammerControllerTest.php
index 4e9e0b3..51d0866 100644
--- a/src/KnpU/CodeBattle/Tests/ProgrammerControllerTest.php
+++ b/src/KnpU/CodeBattle/Tests/ProgrammerControllerTest.php
@@ -23,5 +23,10 @@ class ProgrammerControllerTest extends \PHPUnit_Framework_TestCase
 
         $request = $client->post('/api/programmers', null, json_encode($data));
         $response = $request->send();
+
+        $this->assertEquals(201, $response->getStatusCode());
+        $this->assertTrue($response->hasHeader('Location'));
+        $data = json_decode($response->getBody(true), true);
+        $this->assertArrayHasKey('nickname', $data);
     }
 }
