diff --git a/features/api/ApiFeatureContext.php b/features/api/ApiFeatureContext.php
index c61ebe5..9d71444 100644
--- a/features/api/ApiFeatureContext.php
+++ b/features/api/ApiFeatureContext.php
@@ -100,6 +100,14 @@ class ApiFeatureContext extends BehatContext
     }
 
     /**
+     * @BeforeScenario
+     */
+    public function clearData()
+    {
+        $this->getProjectHelper()->reloadDatabase();
+    }
+
+    /**
      * @Given /^I have the payload:$/
      */
     public function iHaveThePayload(PyStringNode $requestPayload)
