diff --git a/features/api/ApiFeatureContext.php b/features/api/ApiFeatureContext.php
index 80379d3..bb8ef45 100644
--- a/features/api/ApiFeatureContext.php
+++ b/features/api/ApiFeatureContext.php
@@ -79,6 +79,14 @@ class ApiFeatureContext extends BehatContext
     }
 
     /**
+     * @BeforeScenario
+     */
+    public function clearData()
+    {
+        $this->getProjectHelper()->reloadDatabase();
+    }
+
+    /**
      * @Given /^I have the payload:$/
      */
     public function iHaveThePayload(PyStringNode $requestPayload)
