diff --git a/src/KnpU/CodeBattle/Tests/ProgrammerControllerTest.php b/src/KnpU/CodeBattle/Tests/ProgrammerControllerTest.php
index 585ab2d..4e9e0b3 100644
--- a/src/KnpU/CodeBattle/Tests/ProgrammerControllerTest.php
+++ b/src/KnpU/CodeBattle/Tests/ProgrammerControllerTest.php
@@ -1,7 +1,27 @@
 <?php
 namespace KnpU\CodeBattle\Tests;
 
+use Guzzle\Http\Client;
+
 class ProgrammerControllerTest extends \PHPUnit_Framework_TestCase
 {
+    public function testPOST()
+    {
+        // create our http client (Guzzle)
+        $client = new Client('http://localhost:8000', array(
+            'request.options' => array(
+                'exceptions' => false,
+            )
+        ));
+
+        $nickname = 'ObjectOrienter'.rand(0, 999);
+        $data = array(
+            'nickname' => $nickname,
+            'avatarNumber' => 5,
+            'tagLine' => 'a test dev!'
+        );
 
+        $request = $client->post('/api/programmers', null, json_encode($data));
+        $response = $request->send();
+    }
 }
