diff --git a/features/api/ApiFeatureContext.php b/features/api/ApiFeatureContext.php
index 0f7cbc9..e50cd07 100644
--- a/features/api/ApiFeatureContext.php
+++ b/features/api/ApiFeatureContext.php
@@ -82,6 +82,14 @@ class ApiFeatureContext extends BehatContext
     }
 
     /**
+     * @BeforeScenario
+     */
+    public function clearData()
+    {
+        $this->getProjectHelper()->reloadDatabase();
+    }
+
+    /**
      * @Given /^I have the payload:$/
      */
     public function iHaveThePayload(PyStringNode $requestPayload)
