diff --git a/features/api/programmer.feature b/features/api/programmer.feature
index cd43ccd..9b6e52e 100644
--- a/features/api/programmer.feature
+++ b/features/api/programmer.feature
@@ -63,3 +63,10 @@ Feature: Programmer
     And the "avatarNumber" property should equal "2"
     But the "powerLevel" property should equal "4"
     But the "nickname" property should equal "CowboyCoder"
+
+  Scenario: DELETE a programmer
+    Given the following programmers exist:
+      | nickname   | avatarNumber |
+      | UnitTester | 3            |
+    When I request "DELETE /api/programmers/UnitTester"
+    Then the response status code should be 204
