diff --git a/src/KnpU/CodeBattle/Application.php b/src/KnpU/CodeBattle/Application.php
index 2ee65e3..b85d504 100644
--- a/src/KnpU/CodeBattle/Application.php
+++ b/src/KnpU/CodeBattle/Application.php
@@ -250,8 +250,11 @@ class Application extends SilexApplication
                 }
             }
 
+            $data = $apiProblem->toArray();
+            // making type a URL, to a temporarily fake page
+            $data['type'] = 'http://localhost:8000/api/docs/errors#'.$data['type'];
             $response = new JsonResponse(
-                $apiProblem->toArray(),
+                $data,
                 $statusCode
             );
             $response->headers->set('Content-Type', 'application/problem+json');
