diff --git a/src/KnpU/CodeBattle/Application.php b/src/KnpU/CodeBattle/Application.php
index 598d997..d436224 100644
--- a/src/KnpU/CodeBattle/Application.php
+++ b/src/KnpU/CodeBattle/Application.php
@@ -288,6 +288,11 @@ class Application extends SilexApplication
                 return;
             }
 
+            // allow 500 errors in debug to be thrown
+            if ($app['debug'] && $statusCode == 500) {
+                return;
+            }
+
             if ($e instanceof ApiProblemException) {
                 $apiProblem = $e->getApiProblem();
             } else {
