diff --git a/features/api/programmer.feature b/features/api/programmer.feature
index 6db6a25..a52e61c 100644
--- a/features/api/programmer.feature
+++ b/features/api/programmer.feature
@@ -50,6 +50,8 @@ Feature: Programmer
       """
     When I request "POST /api/programmers"
     Then the response status code should be 400
+    And the "Content-Type" header should be "application/problem+json"
+    And the "type" property should equal "invalid_body_format"
 
   Scenario: GET one programmer
     Given the following programmers exist:
