diff --git a/features/api/programmer.feature b/features/api/programmer.feature
index 667ef35..169daca 100644
--- a/features/api/programmer.feature
+++ b/features/api/programmer.feature
@@ -59,6 +59,7 @@ Feature: Programmer
     And the "Content-Type" header should be "application/problem+json"
     And the "type" property should equal "about:blank"
     And the "title" property should equal "Not Found"
+    And the "detail" property should contain "programmer has deserted"
 
   Scenario: GET one programmer
     Given the following programmers exist:
diff --git a/src/KnpU/CodeBattle/Application.php b/src/KnpU/CodeBattle/Application.php
index 9809328..86cfda2 100644
--- a/src/KnpU/CodeBattle/Application.php
+++ b/src/KnpU/CodeBattle/Application.php
@@ -31,6 +31,7 @@ use KnpU\CodeBattle\Repository\ProgrammerRepository;
 use KnpU\CodeBattle\Battle\BattleManager;
 use Silex\Provider\ValidatorServiceProvider;
 use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpKernel\Exception\HttpException;
 use Symfony\Component\Validator\Mapping\ClassMetadataFactory;
 use Symfony\Component\Validator\Mapping\Loader\AnnotationLoader;
 
@@ -303,6 +304,16 @@ class Application extends SilexApplication
                 $apiProblem = new ApiProblem(
                     $statusCode
                 );
+
+                /*
+                 * If it's an HttpException message (e.g. for 404, 403),
+                 * we'll say as a rule that the exception message is safe
+                 * for the client. Otherwise, it could be some sensitive
+                 * low-level exception, which should *not* be exposed
+                 */
+                if ($e instanceof HttpException) {
+                    $apiProblem->set('detail', $e->getMessage());
+                }
             }
 
             $response = new JsonResponse(
