diff --git a/features/api/programmer.feature b/features/api/programmer.feature
index 1d9d6f0..c48538e 100644
--- a/features/api/programmer.feature
+++ b/features/api/programmer.feature
@@ -38,6 +38,7 @@ Feature: Programmer
       """
     And the "errors.nickname" property should exist
     But the "errors.avatarNumber" property should not exist
+    And the "Content-Type" header should be "application/problem+json"
 
   Scenario: GET one programmer
     Given the following programmers exist:
diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index eedabae..7396e78 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -166,6 +166,9 @@ class ProgrammerController extends BaseController
             'errors' => $errors
         );
 
-        return new JsonResponse($data, 400);
+        $response = new JsonResponse($data, 400);
+        $response->headers->set('Content-Type', 'application/problem+json');
+
+        return $response;
     }
 }
