diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index 8175a9a..10b3fb9 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -35,6 +35,18 @@ class ProgrammerController extends BaseController
         $programmer = new Programmer();
         $this->handleRequest($request, $programmer);
 
+        $errors = $this->validate($programmer);
+        if (!empty($errors)) {
+            $data = array(
+                'type' => 'validation_error',
+                'title' => 'There was a validation error',
+                'errors' => $errors
+            );
+
+            return new JsonResponse($data, 400);
+        }
+
+        $this->save($programmer);
         $data = $this->serializeProgrammer($programmer);
         $response = new JsonResponse($data, 201);
         $programmerUrl = $this->generateUrl(
@@ -83,6 +95,7 @@ class ProgrammerController extends BaseController
         }
 
         $this->handleRequest($request, $programmer);
+        $this->save($programmer);
 
         $data = $this->serializeProgrammer($programmer);
 
@@ -135,8 +148,6 @@ class ProgrammerController extends BaseController
         }
 
         $programmer->userId = $this->findUserByUsername('weaverryan')->id;
-
-        $this->save($programmer);
     }
 
     private function serializeProgrammer(Programmer $programmer)
diff --git a/src/KnpU/CodeBattle/Model/Programmer.php b/src/KnpU/CodeBattle/Model/Programmer.php
index e849254..5d0e11f 100644
--- a/src/KnpU/CodeBattle/Model/Programmer.php
+++ b/src/KnpU/CodeBattle/Model/Programmer.php
@@ -11,6 +11,9 @@ class Programmer
     /* All public properties are persisted */
     public $id;
 
+    /**
+     * @Assert\NotBlank(message="Please enter a clever nickname")
+     */
     public $nickname;
 
     /**
