diff --git a/src/KnpU/CodeBattle/Api/ApiProblem.php b/src/KnpU/CodeBattle/Api/ApiProblem.php
index c0725d6..16ec1f2 100644
--- a/src/KnpU/CodeBattle/Api/ApiProblem.php
+++ b/src/KnpU/CodeBattle/Api/ApiProblem.php
@@ -7,6 +7,12 @@ namespace KnpU\CodeBattle\Api;
  */
 class ApiProblem
 {
+    const TYPE_VALIDATION_ERROR = 'validation_error';
+
+    private static $titles = array(
+        self::TYPE_VALIDATION_ERROR => 'There was a validation error'
+    );
+
     private $statusCode;
 
     private $type;
@@ -15,11 +21,16 @@ class ApiProblem
 
     private $extraData = array();
 
-    public function __construct($statusCode, $type, $title)
+    public function __construct($statusCode, $type)
     {
         $this->statusCode = $statusCode;
         $this->type = $type;
-        $this->title = $title;
+
+        if (!isset(self::$titles[$type])) {
+            throw new \InvalidArgumentException('No title for type '.$type);
+        }
+
+        $this->title = self::$titles[$type];
     }
 
     public function toArray()
diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index 931604e..567790f 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -163,8 +163,7 @@ class ProgrammerController extends BaseController
     {
         $apiProblem = new ApiProblem(
             400,
-            'validation_error',
-            'There was a validation error'
+            ApiProblem::TYPE_VALIDATION_ERROR
         );
         $apiProblem->set('errors', $errors);
 
