diff --git a/src/KnpU/CodeBattle/Api/ApiProblem.php b/src/KnpU/CodeBattle/Api/ApiProblem.php
new file mode 100644
index 0000000..c0725d6
--- /dev/null
+++ b/src/KnpU/CodeBattle/Api/ApiProblem.php
@@ -0,0 +1,46 @@
+<?php
+
+namespace KnpU\CodeBattle\Api;
+
+/**
+ * A wrapper for holding data to be used for a application/problem+json response
+ */
+class ApiProblem
+{
+    private $statusCode;
+
+    private $type;
+
+    private $title;
+
+    private $extraData = array();
+
+    public function __construct($statusCode, $type, $title)
+    {
+        $this->statusCode = $statusCode;
+        $this->type = $type;
+        $this->title = $title;
+    }
+
+    public function toArray()
+    {
+        return array_merge(
+            $this->extraData,
+            array(
+                'status' => $this->statusCode,
+                'type' => $this->type,
+                'title' => $this->title,
+            )
+        );
+    }
+
+    public function set($name, $value)
+    {
+        $this->extraData[$name] = $value;
+    }
+
+    public function getStatusCode()
+    {
+        return $this->statusCode;
+    }
+}
diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index 7396e78..931604e 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -2,6 +2,7 @@
 
 namespace KnpU\CodeBattle\Controller\Api;
 
+use KnpU\CodeBattle\Api\ApiProblem;
 use KnpU\CodeBattle\Controller\BaseController;
 use Silex\Application;
 use Silex\ControllerCollection;
@@ -160,13 +161,14 @@ class ProgrammerController extends BaseController
 
     private function handleValidationResponse(array $errors)
     {
-        $data = array(
-            'type' => 'validation_error',
-            'title' => 'There was a validation error',
-            'errors' => $errors
+        $apiProblem = new ApiProblem(
+            400,
+            'validation_error',
+            'There was a validation error'
         );
+        $apiProblem->set('errors', $errors);
 
-        $response = new JsonResponse($data, 400);
+        $response = new JsonResponse($apiProblem->toArray(), $apiProblem->getStatusCode());
         $response->headers->set('Content-Type', 'application/problem+json');
 
         return $response;
