diff --git a/features/api/programmer.feature b/features/api/programmer.feature
index 169daca..58e72be 100644
--- a/features/api/programmer.feature
+++ b/features/api/programmer.feature
@@ -51,7 +51,7 @@ Feature: Programmer
     When I request "POST /api/programmers"
     Then the response status code should be 400
     And the "Content-Type" header should be "application/problem+json"
-    And the "type" property should equal "invalid_body_format"
+    And the "type" property should contain "invalid_body_format"
 
   Scenario: Proper 404 exception on no programmer
     When I request "GET /api/programmers/fake"
diff --git a/src/KnpU/CodeBattle/Application.php b/src/KnpU/CodeBattle/Application.php
index 1769d4f..833102b 100644
--- a/src/KnpU/CodeBattle/Application.php
+++ b/src/KnpU/CodeBattle/Application.php
@@ -321,8 +321,13 @@ class Application extends SilexApplication
                 }
             }
 
+            $data = $apiProblem->toArray();
+            // making type a URL, to a temporarily fake page
+            if ($data['type'] != 'about:blank') {
+                $data['type'] = 'http://localhost:8000/docs/errors#'.$data['type'];
+            }
             $response = new JsonResponse(
-                $apiProblem->toArray(),
+                $data,
                 $apiProblem->getStatusCode()
             );
             $response->headers->set('Content-Type', 'application/problem+json');
