diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index 200ab4a..f6ed37d 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -15,6 +15,8 @@ class ProgrammerController extends BaseController
     protected function addRoutes(ControllerCollection $controllers)
     {
         $controllers->post('/api/programmers', array($this, 'newAction'));
+
+        $controllers->get('/api/programmers/{nickname}', array($this, 'showAction'));
     }
 
     public function newAction(Request $request)
@@ -32,4 +34,9 @@ class ProgrammerController extends BaseController
 
         return $response;
     }
+
+    public function showAction($nickname)
+    {
+        return new Response('Hello '.$nickname);
+    }
 }
diff --git a/testing.php b/testing.php
index 2cabb17..0d8e288 100644
--- a/testing.php
+++ b/testing.php
@@ -18,8 +18,13 @@ $data = array(
     'tagLine' => 'a test dev!'
 );
 
+// 1) Create a programmer resource
 $request = $client->post('/api/programmers', null, json_encode($data));
 $response = $request->send();
 
+// 2) GET a programmer resource
+$request = $client->get('/api/programmers/'.$nickname);
+$response = $request->send();
+
 echo $response;
 echo "\n\n";
\ No newline at end of file
