diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index 355f348..0315d3c 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -18,6 +18,8 @@ class ProgrammerController extends BaseController
 
         $controllers->get('/api/programmers/{nickname}', array($this, 'showAction'))
             ->bind('api_programmers_show');
+
+        $controllers->get('/api/programmers', array($this, 'listAction'));
     }
 
     public function newAction(Request $request)
@@ -49,15 +51,34 @@ class ProgrammerController extends BaseController
             $this->throw404();
         }
 
-        $data = array(
-            'nickname' => $programmer->nickname,
-            'avatarNumber' => $programmer->avatarNumber,
-            'powerLevel' => $programmer->powerLevel,
-        );
+        $data = $this->serializeProgrammer($programmer);
 
         $response = new Response(json_encode($data), 200);
         $response->headers->set('Content-Type', 'application/json');
 
         return $response;
     }
+
+    public function listAction()
+    {
+        $programmers = $this->getProgrammerRepository()->findAll();
+        $data = ['data' => []];
+        foreach ($programmers as $programmer) {
+            $data['data'][] = $this->serializeProgrammer($programmer);
+        }
+
+        $response = new Response(json_encode($data), 200);
+        $response->headers->set('Content-Type', 'application/json');
+
+        return $response;
+    }
+
+    private function serializeProgrammer(Programmer $programmer)
+    {
+        return array(
+            'nickname' => $programmer->nickname,
+            'avatarNumber' => $programmer->avatarNumber,
+            'powerLevel' => $programmer->powerLevel,
+        );
+    }
 }
diff --git a/testing.php b/testing.php
index 703ca16..ecd1be2 100644
--- a/testing.php
+++ b/testing.php
@@ -27,5 +27,9 @@ $programmerUrl = $response->getHeader('Location');
 $request = $client->get($programmerUrl);
 $response = $request->send();
 
+// 3) GET a list of all programmers
+$request = $client->get('/api/programmers');
+$response = $request->send();
+
 echo $response;
 echo "\n\n";
\ No newline at end of file
