diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index 8c48f32..9a781d5 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -38,6 +38,14 @@ class ProgrammerController extends BaseController
 
     public function showAction($nickname)
     {
-        return new Response('Hello '.$nickname);
+        $programmer = $this->getProgrammerRepository()->findOneByNickname($nickname);
+
+        $data = array(
+            'nickname' => $programmer->nickname,
+            'avatarNumber' => $programmer->avatarNumber,
+            'powerLevel' => $programmer->powerLevel,
+        );
+
+        return new Response(json_encode($data), 200);
     }
 }
