diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index ab69656..67502fd 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -32,12 +32,17 @@ class ProgrammerController extends BaseController
 
         $this->save($programmer);
 
-        $response = new Response('It worked. Believe me - I\'m an API', 201);
+        $data = $this->serializeProgrammer($programmer);
+        $response = new Response(
+            json_encode($data),
+            201
+        );
         $programmerUrl = $this->generateUrl(
             'api_programmers_show',
             ['nickname' => $programmer->nickname]
         );
         $response->headers->set('Location', $programmerUrl);
+        $response->headers->set('Content-Type', 'application/json');
 
         return $response;
     }
diff --git a/testing.php b/testing.php
index 55d06c6..26b82f8 100644
--- a/testing.php
+++ b/testing.php
@@ -22,6 +22,9 @@ $data = array(
 $request = $client->post('/api/programmers', null, json_encode($data));
 $response = $request->send();
 
+echo $response;
+echo "\n\n";die;
+
 $programmerUrl = $response->getHeader('Location');
 
 // 2) GET a programmer resource
