diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index 67502fd..d1fc062 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -33,16 +33,12 @@ class ProgrammerController extends BaseController
         $this->save($programmer);
 
         $data = $this->serializeProgrammer($programmer);
-        $response = new Response(
-            json_encode($data),
-            201
-        );
+        $response = new JsonResponse($data, 201);
         $programmerUrl = $this->generateUrl(
             'api_programmers_show',
             ['nickname' => $programmer->nickname]
         );
         $response->headers->set('Location', $programmerUrl);
-        $response->headers->set('Content-Type', 'application/json');
 
         return $response;
     }
@@ -57,8 +53,7 @@ class ProgrammerController extends BaseController
 
         $data = $this->serializeProgrammer($programmer);
 
-        $response = new Response(json_encode($data), 200);
-        $response->headers->set('Content-Type', 'application/json');
+        $response = new JsonResponse($data, 200);
 
         return $response;
     }
@@ -71,8 +66,7 @@ class ProgrammerController extends BaseController
             $data['programmers'][] = $this->serializeProgrammer($programmer);
         }
 
-        $response = new Response(json_encode($data), 200);
-        $response->headers->set('Content-Type', 'application/json');
+        $response = new JsonResponse($data, 200);
 
         return $response;
     }
