diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index 61bad86..7a9bedc 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -2,6 +2,7 @@
 
 namespace KnpU\CodeBattle\Controller\Api;
 
+use Hateoas\Representation\CollectionRepresentation;
 use KnpU\CodeBattle\Api\ApiProblem;
 use KnpU\CodeBattle\Api\ApiProblemException;
 use KnpU\CodeBattle\Controller\BaseController;
@@ -72,9 +73,14 @@ class ProgrammerController extends BaseController
     public function listAction()
     {
         $programmers = $this->getProgrammerRepository()->findAll();
-        $data = array('programmers' => $programmers);
 
-        $response = $this->createApiResponse($data, 200, 'json');
+        $collection = new CollectionRepresentation(
+            $programmers,
+            'programmers',
+            'programmers'
+        );
+
+        $response = $this->createApiResponse($collection, 200, 'json');
 
         return $response;
     }
