diff --git a/features/api/battle.feature b/features/api/battle.feature
index 5111da1..e674da4 100644
--- a/features/api/battle.feature
+++ b/features/api/battle.feature
@@ -32,3 +32,4 @@ Feature:
     And the "notes" property should exist
     And the "didProgrammerWin" property should exist
     And the link "programmer" should exist and its value should be "/api/programmers/Fred"
+    And the "Content-Type" header should be "application/hal+json"
diff --git a/src/KnpU/CodeBattle/Controller/BaseController.php b/src/KnpU/CodeBattle/Controller/BaseController.php
index a9ebfe9..5e465a4 100644
--- a/src/KnpU/CodeBattle/Controller/BaseController.php
+++ b/src/KnpU/CodeBattle/Controller/BaseController.php
@@ -239,7 +239,7 @@ abstract class BaseController implements ControllerProviderInterface
         $json = $this->serialize($data);
 
         return new Response($json, $statusCode, array(
-            'Content-Type' => 'application/json'
+            'Content-Type' => 'application/hal+json'
         ));
     }
 
