diff --git a/features/api/battle.feature b/features/api/battle.feature
index f7288fa..71a10dc 100644
--- a/features/api/battle.feature
+++ b/features/api/battle.feature
@@ -32,5 +32,5 @@ Feature:
     And the "notes" property should exist
     And the "didProgrammerWin" property should exist
     And the link "programmer" should exist and its value should be "/api/programmers/Fred"
-    And the "_embedded.programmer.nickname" property should equal "Fred"
+    And the embedded "programmer" should have a "nickname" property equal to "Fred"
     And the "Content-Type" header should be "application/hal+json"
