diff --git a/src/KnpU/CodeBattle/Model/Homepage.php b/src/KnpU/CodeBattle/Model/Homepage.php
index 5c330be..1e7f9c5 100644
--- a/src/KnpU/CodeBattle/Model/Homepage.php
+++ b/src/KnpU/CodeBattle/Model/Homepage.php
@@ -17,4 +17,5 @@ use Hateoas\Configuration\Annotation as Hateoas;
  */
 class Homepage
 {
+    private $message = 'Welcome to the CodeBattles API! Look around at the _links to browse the API. And have a crazy-cool day.';
 }
