diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index d40305e..cc87d0d 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -19,6 +19,9 @@ class ProgrammerController extends BaseController
 {
     protected function addRoutes(ControllerCollection $controllers)
     {
+        // the homepage - put in this controller for simplicity
+        $controllers->get('api', array($this, 'homepageAction'));
+
         $controllers->post('/api/programmers', array($this, 'newAction'));
 
         $controllers->get('/api/programmers/{nickname}', array($this, 'showAction'))
@@ -38,6 +41,11 @@ class ProgrammerController extends BaseController
             ->bind('api_programmers_battles_list');
     }
 
+    public function homepageAction()
+    {
+        return $this->createApiResponse(array());
+    }
+
     public function newAction(Request $request)
     {
         $this->enforceUserSecurity();
