diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index 628d636..5cd4172 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -8,6 +8,7 @@ use Silex\ControllerCollection;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpFoundation\JsonResponse;
+use KnpU\CodeBattle\Model\Programmer;
 
 class ProgrammerController extends BaseController
 {
@@ -18,7 +19,14 @@ class ProgrammerController extends BaseController
 
     public function newAction(Request $request)
     {
-        $data = $request->getContent();
-        return $data;
+        $data = json_decode($request->getContent(), true);
+
+        $programmer = new Programmer($data['nickname'], $data['avatarNumber']);
+        $programmer->tagLine = $data['tagLine'];
+        $programmer->userId = $this->findUserByUsername('weaverryan')->id;
+
+        $this->save($programmer);
+
+        return 'It worked. Believe me - I\'m an API';
     }
 }
