diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index 5cd4172..9175264 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -27,6 +27,6 @@ class ProgrammerController extends BaseController
 
         $this->save($programmer);
 
-        return 'It worked. Believe me - I\'m an API';
+        return new Response('It worked. Believe me - I\'m an API', 201);
     }
 }
