diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index a979f96..0109fa1 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -15,8 +15,9 @@ class ProgrammerController extends BaseController
         $controllers->post('/api/programmers', array($this, 'newAction'));
     }
 
-    public function newAction()
+    public function newAction(Request $request)
     {
-        return 'hello world!';
+        $data = $request->getContent();
+        return $data;
     }
 }
