diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index 3d4fb1b..1abe527 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -3,6 +3,7 @@
 namespace KnpU\CodeBattle\Controller\Api;
 
 use KnpU\CodeBattle\Controller\BaseController;
+use KnpU\CodeBattle\Model\Programmer;
 use Silex\Application;
 use Silex\ControllerCollection;
 use Symfony\Component\HttpFoundation\Request;
@@ -16,7 +17,15 @@ class ProgrammerController extends BaseController
 
     public function newAction(Request $request)
     {
-        $data = $request->getContent();
-        return $data;
+        $data = json_decode($request->getContent(), true);
+
+        $programmer = new Programmer();
+        $programmer->nickname = $data['nickname'];
+        $programmer->avatar = $data['avatar'];
+        $programmer->userId = $this->findUserByUsername('weaverryan')->id;
+
+        $this->save($programmer);
+
+        return 'It worked. Believe me - I\'m an API';
     }
 }
