diff --git a/src/KnpU/CodeBattle/Controller/Api/BattleController.php b/src/KnpU/CodeBattle/Controller/Api/BattleController.php
index de046e3..d8573d5 100644
--- a/src/KnpU/CodeBattle/Controller/Api/BattleController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/BattleController.php
@@ -11,7 +11,8 @@ class BattleController extends BaseController
     protected function addRoutes(ControllerCollection $controllers)
     {
         $controllers->post('/api/battles', array($this, 'newAction'));
-        $controllers->get('/api/battles/{id}', array($this, 'showAction'));
+        $controllers->get('/api/battles/{id}', array($this, 'showAction'))
+            ->bind('api_battles_show');
     }
 
     public function newAction(Request $request)
diff --git a/src/KnpU/CodeBattle/Model/Battle.php b/src/KnpU/CodeBattle/Model/Battle.php
index 08bef5b..20a5ab5 100644
--- a/src/KnpU/CodeBattle/Model/Battle.php
+++ b/src/KnpU/CodeBattle/Model/Battle.php
@@ -8,6 +8,13 @@ use Hateoas\Configuration\Annotation as Hateoas;
 
 /**
  * @Hateoas\Relation(
+ *      "self",
+ *      href = @Hateoas\Route(
+ *          "api_battles_show",
+ *          parameters = { "id" = "expr(object.id)" }
+ *      )
+ * )
+ * @Hateoas\Relation(
  *      "programmer",
  *      href = @Hateoas\Route(
  *          "api_programmers_show",
