diff --git a/src/KnpU/CodeBattle/Model/Battle.php b/src/KnpU/CodeBattle/Model/Battle.php
index 08bef5bb..86218afc 100644
--- a/src/KnpU/CodeBattle/Model/Battle.php
+++ b/src/KnpU/CodeBattle/Model/Battle.php
@@ -7,6 +7,13 @@ use JMS\Serializer\Annotation as Serializer;
 use Hateoas\Configuration\Annotation as Hateoas;
 
 /**
+ * @Hateoas\Relation(
+ *      "self",
+ *      href = @Hateoas\Route(
+ *          "api_battle_show",
+ *          parameters = { "nickname" = "expr(object.id)" }
+ *      )
+ * )
  * @Hateoas\Relation(
  *      "programmer",
  *      href = @Hateoas\Route(
