diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index 7a9bedc..d40305e 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -33,6 +33,9 @@ class ProgrammerController extends BaseController
             ->method('PATCH');
 
         $controllers->delete('/api/programmers/{nickname}', array($this, 'deleteAction'));
+
+        $controllers->get('/api/programmers/{nickname}/battles', array($this, 'listBattlesAction'))
+            ->bind('api_programmers_battles_list');
     }
 
     public function newAction(Request $request)
@@ -85,6 +88,28 @@ class ProgrammerController extends BaseController
         return $response;
     }
 
+    public function listBattlesAction($nickname)
+    {
+        $programmer = $this->getProgrammerRepository()->findOneByNickname($nickname);
+
+        if (!$programmer) {
+            $this->throw404('Oh no! This programmer has deserted! We\'ll send a search party!');
+        }
+
+        $battles = $this->getBattleRepository()
+            ->findAllBy(array('programmerId' => $programmer->id));
+
+        $collection = new CollectionRepresentation(
+            $battles,
+            'battles',
+            'battles'
+        );
+
+        $response = $this->createApiResponse($collection);
+
+        return $response;
+    }
+
     public function updateAction($nickname, Request $request)
     {
         $programmer = $this->getProgrammerRepository()->findOneByNickname($nickname);
