diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index a12fe3e..12ffd1f 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -25,14 +25,8 @@ class ProgrammerController extends BaseController
 
     public function newAction(Request $request)
     {
-        $data = json_decode($request->getContent(), true);
-
         $programmer = new Programmer();
-        $programmer->nickname = $data['nickname'];
-        $programmer->avatarNumber = $data['avatarNumber'];
-        $programmer->tagLine = $data['tagLine'];
-        $programmer->userId = $this->findUserByUsername('weaverryan')->id;
-
+        $this->updateProgrammerFromRequest($request, $programmer);
         $this->save($programmer);
 
         $data = $this->serializeProgrammer($programmer);
@@ -82,13 +76,7 @@ class ProgrammerController extends BaseController
             $this->throw404();
         }
 
-        $data = json_decode($request->getContent(), true);
-
-        $programmer->nickname = $data['nickname'];
-        $programmer->avatarNumber = $data['avatarNumber'];
-        $programmer->tagLine = $data['tagLine'];
-        $programmer->userId = $this->findUserByUsername('weaverryan')->id;
-
+        $this->updateProgrammerFromRequest($request, $programmer);
         $this->save($programmer);
 
         $data = $this->serializeProgrammer($programmer);
@@ -98,6 +86,26 @@ class ProgrammerController extends BaseController
         return $response;
     }
 
+    /**
+     * Reads data from the Request, updates the Programmer and saves it.
+     *
+     * @param Request $request
+     * @param Programmer $programmer
+     */
+    private function updateProgrammerFromRequest(Request $request, Programmer $programmer)
+    {
+        $data = json_decode($request->getContent(), true);
+
+        if ($data === null) {
+            throw new \Exception(sprintf('Invalid JSON: '.$request->getContent()));
+        }
+
+        $programmer->nickname = $data['nickname'];
+        $programmer->avatarNumber = $data['avatarNumber'];
+        $programmer->tagLine = $data['tagLine'];
+        $programmer->userId = $this->findUserByUsername('weaverryan')->id;
+    }
+
     private function serializeProgrammer(Programmer $programmer)
     {
         return array(
