diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index d1fc062..b3c7c1f 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -20,6 +20,7 @@ class ProgrammerController extends BaseController
             ->bind('api_programmers_show');
 
         $controllers->get('/api/programmers', array($this, 'listAction'));
+        $controllers->put('/api/programmers/{nickname}', array($this, 'updateAction'));
     }
 
     public function newAction(Request $request)
@@ -71,6 +72,30 @@ class ProgrammerController extends BaseController
         return $response;
     }
 
+    public function updateAction($nickname, Request $request)
+    {
+        $programmer = $this->getProgrammerRepository()->findOneByNickname($nickname);
+
+        if (!$programmer) {
+            $this->throw404('Oh no! This programmer has deserted! We\'ll send a search party!');
+        }
+
+        $data = json_decode($request->getContent(), true);
+
+        $programmer->nickname = $data['nickname'];
+        $programmer->avatarNumber = $data['avatarNumber'];
+        $programmer->tagLine = $data['tagLine'];
+        $programmer->userId = $this->findUserByUsername('weaverryan')->id;
+
+        $this->save($programmer);
+
+        $data = $this->serializeProgrammer($programmer);
+
+        $response = new JsonResponse($data, 200);
+
+        return $response;
+    }
+
     private function serializeProgrammer(Programmer $programmer)
     {
         return array(
