diff --git a/features/api/programmer.feature b/features/api/programmer.feature
index 5980364..0c889cd 100644
--- a/features/api/programmer.feature
+++ b/features/api/programmer.feature
@@ -52,7 +52,7 @@ Feature: Programmer
     And I have the payload:
       """
       {
-        "nickname": "CowboyCoder",
+        "nickname": "CowgirlCoder",
         "avatarNumber" : 2,
         "tagLine": "foo"
       }
@@ -60,3 +60,4 @@ Feature: Programmer
     When I request "PUT /api/programmers/CowboyCoder"
     Then the response status code should be 200
     And the "avatarNumber" property should equal "2"
+    But the "nickname" property should equal "CowboyCoder"
diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index 1f34f08..b2e7ec9 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -93,13 +93,17 @@ class ProgrammerController extends BaseController
     private function handleRequest(Request $request, Programmer $programmer)
     {
         $data = json_decode($request->getContent(), true);
+        $isNew = !$programmer->id;
 
         if ($data === null) {
             throw new \Exception(sprintf('Invalid JSON: '.$request->getContent()));
         }
 
         // determine which properties should be changeable on this request
-        $apiProperties = array('nickname', 'avatarNumber', 'tagLine');
+        $apiProperties = array('avatarNumber', 'tagLine');
+        if ($isNew) {
+            $apiProperties[] = 'nickname';
+        }
 
         // update the properties
         foreach ($apiProperties as $property) {
