diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index 3349be0..5891057 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -25,15 +25,8 @@ class ProgrammerController extends BaseController
 
     public function newAction(Request $request)
     {
-        $data = json_decode($request->getContent(), true);
-
         $programmer = new Programmer();
-        $programmer->nickname = $data['nickname'];
-        $programmer->avatarNumber = $data['avatarNumber'];
-        $programmer->tagLine = $data['tagLine'];
-        $programmer->userId = $this->findUserByUsername('weaverryan')->id;
-
-        $this->save($programmer);
+        $this->updateProgrammerFromRequest($request, $programmer);
 
         $data = $this->serializeProgrammer($programmer);
         $response = new JsonResponse($data, 201);
@@ -82,20 +75,35 @@ class ProgrammerController extends BaseController
             $this->throw404();
         }
 
+        $this->updateProgrammerFromRequest($request, $programmer);
+
+        $data = $this->serializeProgrammer($programmer);
+
+        $response = new JsonResponse($data, 200);
+
+        return $response;
+    }
+
+    /**
+     * Reads data from the Request, updates the Programmer and saves it.
+     *
+     * @param Request $request
+     * @param Programmer $programmer
+     */
+    private function updateProgrammerFromRequest(Request $request, Programmer $programmer)
+    {
         $data = json_decode($request->getContent(), true);
 
+        if ($data === null) {
+            throw new \Exception(sprintf('Invalid JSON: '.$request->getContent()));
+        }
+
         $programmer->nickname = $data['nickname'];
         $programmer->avatarNumber = $data['avatarNumber'];
         $programmer->tagLine = $data['tagLine'];
         $programmer->userId = $this->findUserByUsername('weaverryan')->id;
 
         $this->save($programmer);
-
-        $data = $this->serializeProgrammer($programmer);
-
-        $response = new JsonResponse($data, 200);
-
-        return $response;
     }
 
     private function serializeProgrammer(Programmer $programmer)
