diff --git a/features/api/programmer.feature b/features/api/programmer.feature
index 51efc15..cd43ccd 100644
--- a/features/api/programmer.feature
+++ b/features/api/programmer.feature
@@ -47,16 +47,19 @@ Feature: Programmer
 
   Scenario: PUT to update a programmer
     Given the following programmers exist:
-      | nickname    | avatarNumber | tagLine |
-      | CowboyCoder | 5            | foo     |
+      | nickname    | avatarNumber | tagLine | powerLevel |
+      | CowboyCoder | 5            | foo     | 4          |
     And I have the payload:
       """
       {
-        "nickname": "CowboyCoder",
+        "nickname": "CowgirlCoder",
         "avatarNumber" : 2,
-        "tagLine": "foo"
+        "tagLine": "foo",
+        "powerLevel": 10
       }
       """
     When I request "PUT /api/programmers/CowboyCoder"
     Then the response status code should be 200
     And the "avatarNumber" property should equal "2"
+    But the "powerLevel" property should equal "4"
+    But the "nickname" property should equal "CowboyCoder"
diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index 017c6cd..3dd031d 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -95,14 +95,24 @@ class ProgrammerController extends BaseController
     private function updateProgrammerFromRequest(Request $request, Programmer $programmer)
     {
         $data = json_decode($request->getContent(), true);
+        $isNew = !$programmer->id;
 
         if ($data === null) {
             throw new \Exception(sprintf('Invalid JSON: '.$request->getContent()));
         }
 
-        $programmer->nickname = $data['nickname'];
-        $programmer->avatarNumber = $data['avatarNumber'];
-        $programmer->tagLine = $data['tagLine'];
+        // determine which properties should be changeable on this request
+        $apiProperties = array('avatarNumber', 'tagLine');
+        if ($isNew) {
+            $apiProperties[] = 'nickname';
+        }
+
+        // update the properties
+        foreach ($apiProperties as $property) {
+            $val = isset($data[$property]) ? $data[$property] : null;
+            $programmer->$property = $val;
+        }
+
         $programmer->userId = $this->findUserByUsername('weaverryan')->id;
     }
 
