diff --git a/features/api/programmer.feature b/features/api/programmer.feature
index 1f0c4dc..5980364 100644
--- a/features/api/programmer.feature
+++ b/features/api/programmer.feature
@@ -44,3 +44,19 @@ Feature: Programmer
     Then the response status code should be 200
     And the "programmers" property should be an array
     And the "programmers" property should contain 2 items
+
+  Scenario: PUT to update a programmer
+    Given the following programmers exist:
+      | nickname    | avatarNumber | tagLine |
+      | CowboyCoder | 5            | foo     |
+    And I have the payload:
+      """
+      {
+        "nickname": "CowboyCoder",
+        "avatarNumber" : 2,
+        "tagLine": "foo"
+      }
+      """
+    When I request "PUT /api/programmers/CowboyCoder"
+    Then the response status code should be 200
+    And the "avatarNumber" property should equal "2"
