diff --git a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
index d79543a..741222e 100644
--- a/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
+++ b/src/KnpU/CodeBattle/Controller/Api/ProgrammerController.php
@@ -43,8 +43,8 @@ class ProgrammerController extends BaseController
         }
 
         $this->save($programmer);
-        $data = $this->serializeProgrammer($programmer);
-        $response = new JsonResponse($data, 201);
+        $json = $this->serialize($programmer);
+        $response = new Response($json, 201);
         $programmerUrl = $this->generateUrl(
             'api_programmers_show',
             ['nickname' => $programmer->nickname]
@@ -62,9 +62,9 @@ class ProgrammerController extends BaseController
             $this->throw404('Oh no! This programmer has deserted! We\'ll send a search party!');
         }
 
-        $data = $this->serializeProgrammer($programmer);
+        $json = $this->serialize($programmer);
 
-        $response = new JsonResponse($data, 200);
+        $response = new Response($json, 200);
 
         return $response;
     }
@@ -72,12 +72,10 @@ class ProgrammerController extends BaseController
     public function listAction()
     {
         $programmers = $this->getProgrammerRepository()->findAll();
-        $data = array('programmers' => array());
-        foreach ($programmers as $programmer) {
-            $data['programmers'][] = $this->serializeProgrammer($programmer);
-        }
+        $data = array('programmers' => $programmers);
+        $json = $this->serialize($data);
 
-        $response = new JsonResponse($data, 200);
+        $response = new Response($json, 200);
 
         return $response;
     }
@@ -98,9 +96,9 @@ class ProgrammerController extends BaseController
 
         $this->save($programmer);
 
-        $data = $this->serializeProgrammer($programmer);
+        $json = $this->serialize($programmer);
 
-        $response = new JsonResponse($data, 200);
+        $response = new Response($json, 200);
 
         return $response;
     }
@@ -155,14 +153,9 @@ class ProgrammerController extends BaseController
         $programmer->userId = $this->findUserByUsername('weaverryan')->id;
     }
 
-    private function serializeProgrammer(Programmer $programmer)
+    protected function serialize($data)
     {
-        return array(
-            'nickname' => $programmer->nickname,
-            'avatarNumber' => $programmer->avatarNumber,
-            'powerLevel' => $programmer->powerLevel,
-            'tagLine' => $programmer->tagLine,
-        );
+        return $this->container['serializer']->serialize($data, 'json');
     }
 
     private function throwApiProblemValidationException(array $errors)
