diff --git a/src/KnpU/CodeBattle/Controller/BaseController.php b/src/KnpU/CodeBattle/Controller/BaseController.php
index 584893e..9d6cc57 100644
--- a/src/KnpU/CodeBattle/Controller/BaseController.php
+++ b/src/KnpU/CodeBattle/Controller/BaseController.php
@@ -2,6 +2,7 @@
 
 namespace KnpU\CodeBattle\Controller;
 
+use JMS\Serializer\SerializationContext;
 use KnpU\CodeBattle\Model\Programmer;
 use KnpU\CodeBattle\Model\User;
 use KnpU\CodeBattle\Repository\UserRepository;
@@ -240,6 +241,9 @@ abstract class BaseController implements ControllerProviderInterface
 
     protected function serialize($data, $format = 'json')
     {
-        return $this->container['serializer']->serialize($data, $format);
+        $context = new SerializationContext();
+        $context->setSerializeNull(true);
+
+        return $this->container['serializer']->serialize($data, $format, $context);
     }
 }
